File: VB/SpreadWinDemo/rowcolcell/rowheaderautowidth.vb
Public Class rowheaderautowidth

    Public Sub New()

        InitializeComponent()

        ' SPREADの設定
        InitSpread(FpSpread1)
    End Sub

    Private Sub InitSpread(spread As FarPoint.Win.Spread.FpSpread)
        FpSpread1.ActiveSheet.RowCount = 1000000
    End Sub

    Private Sub CheckBox1_CheckedChanged(sender As Object, e As System.EventArgs) Handles CheckBox1.CheckedChanged
        If CheckBox1.Checked Then
            ' 行ヘッダの自動拡張を有効にする
            FpSpread1.RowHeaderAutoWidthMax = -1
        Else
            ' 行ヘッダの自動拡張を有効にしない
            FpSpread1.RowHeaderAutoWidthMax = 0
        End If
    End Sub
End Class

File: CS/SpreadWinDemoCS/rowcolcell/rowheaderautowidth.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace SpreadWinDemo.rowcolcell
{
    public partial class rowheaderautowidth : SpreadWinDemo.DemoBase
    {
        public rowheaderautowidth()
        {
            InitializeComponent();

            // SPREADの設定
            InitSpread(fpSpread1);

            checkBox1.CheckedChanged += new EventHandler(checkBox1_CheckedChanged);
        }

        private void InitSpread(FarPoint.Win.Spread.FpSpread spread)
        {
            fpSpread1.ActiveSheet.RowCount = 1000000;
        }

        void checkBox1_CheckedChanged(object sender, EventArgs e)
        {
            if (checkBox1.Checked)
            {
                // 行ヘッダの自動拡張を有効にする
                fpSpread1.RowHeaderAutoWidthMax = -1;
            }
            else
            {
                // 行ヘッダの自動拡張を有効にしない
                fpSpread1.RowHeaderAutoWidthMax = 0;
            }
        }
    }
}
